src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs

```diff
@@ -80,17 +81,75 @@ private async Task ExecuteAsync()
             }
         }
 
-        private string GetDownloadUrl()
+        private async Task<string> GetDownloadUrlAsync()
         {
             string extension = Runtime.StartsWith("win") ? "zip" : "tar.gz";
+            string effectiveVersion = await GetEffectiveVersion();
+
             return PackageType switch
             {
-                "sdk"                => $"{DotNetCliAzureFeed}/Sdk/{Version}/dotnet-sdk-{Version}-{Runtime}.{extension}",
-                "aspnetcore-runtime" => $"{DotNetCliAzureFeed}/aspnetcore/Runtime/{Version}/aspnetcore-runtime-{Version}-{Runtime}.{extension}",
-                _                    => $"{DotNetCliAzureFeed}/Runtime/{Version}/dotnet-runtime-{Version}-{Runtime}.{extension}"
+                "sdk" => $"{DotNetCliAzureFeed}/Sdk/{Version}/dotnet-sdk-{effectiveVersion}-{Runtime}.{extension}",
+                "aspnetcore-runtime" => $"{DotNetCliAzureFeed}/aspnetcore/Runtime/{Version}/aspnetcore-runtime-{effectiveVersion}-{Runtime}.{extension}",
+                _ => $"{DotNetCliAzureFeed}/Runtime/{Version}/dotnet-runtime-{effectiveVersion}-{Runtime}.{extension}"
             };
         }
 
+        private async Task<string> GetEffectiveVersion()
+        {
+            if (NuGetVersion.TryParse(Version, out NuGetVersion semanticVersion))
+            {
+                // Pared down version of the logic from https://github.com/dotnet/install-scripts/blob/main/src/dotnet-install.ps1
+                // If this functionality stops working, review changes made there.
+                // Current strategy is to start with a runtime-specific name then fall back to 'productVersion.txt'
+                string effectiveVersion = Version;
+
+                // Do nothing for older runtimes; the file won't exist
+                if (semanticVersion > new NuGetVersion("3.1.0"))
```

Comment:
Suggest
```suggestion
                if (semanticVersion >= new NuGetVersion("5.0.0"))
```

Otherwise
- Won't this incorrectly try to download runtime assets for `3.1.1` and later❔
- Is this the correct check for SDK versions e.g. `3.1.412`❔

Reply:
good point, will do.